auth/authentication.md

```diff
@@ -0,0 +1,61 @@
+# Authentication
+
+## User Flow
+
+User visits a new website that requires authentication from wallet.
+
+1. Website displays qrcode or deep link for auth
+2. User scans qrcode or redirects to wallet
+3. User approves prompt for auth in wallet
+4. User returns to website after prompt success
+5. Website is now authenticated with signed message
+
+User returns to a previously visted website before authentication expiry
+
+1. Website checks if user is authenticated from local storage
+2. Website is now authenticated with signed message
+
+User returns to a previously visted website after authentication expiry
+
+1. Website checks if user is authenticated from local storage
+2. Website sends new request to previously paired wallet
+3. User views notification and redirects to wallet
```

Comment:
If I understand correctly would it require a push support on mobile devices?

Reply:
Not necessarily a requirement because you have a websocket on the wallet side but it does improve the UX

Reply:
Websockets / long polling require app to be in foreground. Push notifications is the only way to make it properly. Otherwise we can ask user to scan QR again

Reply:
The pattern isn't that different from Sign API.

User is either prompted or redirected to the Wallet to approve the authentication signature

The Push Notification would be a nice-to-have

However I can see how it's adding confusion to the spec... I will rephrase it